### PR TITLE
feat(api): add event model

### PR DESCRIPTION
**Pull request by PranshuNayak**
@Samy-33 raised PR

### CODE REVIEW

File: api/api/events/models.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from django.db import models
+
+
+class Events(models.Model):
+    name = models.CharField(max_length=100)
+    description = models.TextField()
+    starts_at = models.DateField()
+    ends_at = models.DateField()
+    registration_required = models.BooleanField(default=False)
+    created_by = models.OneToOneField(User, on_delete=models.PROTECT)
```
Comment (Samy-33) [inline api/api/events/models.py:12]: `created_by` should be set `null` if the user is deleted. For that you'd need to make this field nullable as well.

File: api/api/events/models.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    image_url = models.URLField(max_length=500)
+    club = models.OneToOneField(Club, on_delete=models.PROTECT)
```
Comment (Samy-33) [inline api/api/events/models.py:14]: Add a comment on top of this: `This club is the primary organizer of this event.`

File: api/api/events/models.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    registration_approval_required = models.BooleanField(default=False)
+    address = models.TextField()
```
Comment (Samy-33) [inline api/api/events/models.py:16]: Why can't it be a `CharField`? `max_length` could be 500.

**Review by Samy-33 — CHANGES_REQUESTED** [review]

**Comment by Samy-33** [comment]
When you need the PR to be reviewed, don't leave it in `Draft` mode. Mark it ready for review.

File: api/api/events/models.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+from django.db import models
+
+
+class Events(models.Model):
+    name = models.CharField(max_length=100)
+    description = models.TextField()
+    starts_at = models.DateField()
+    ends_at = models.DateField()
+    registration_required = models.BooleanField(default=False)
+    created_by = models.OneToOneField(User, null=True, on_delete=models.SET_NULL)
```
Comment (Samy-33) [inline api/api/events/models.py:12]: `created_by` should be a `ForeignKeyField`.

Add `related_name` to all the `ForeignKeys`, `OneToOne` and `OneToMany` relationships.

File: api/api/events/models.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    image_url = models.URLField(max_length=500)
+    # This club is the primary organizer of this event
+    club = models.OneToOneField(Club, on_delete=models.PROTECT)
```
Comment (Samy-33) [inline api/api/events/models.py:15]: should be a `ForeignKeyField`

**Review by Samy-33 — CHANGES_REQUESTED** [review]

File: api/api/events/models.py (hunk 4 above)
Comment (PranshuNayak) [reply api/api/events/models.py:12]: for `created_by` related name `events_created` works ?

**Review by Samy-33 — APPROVED** [review]
LGTM

**Pull request merged**